b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/createRoleComponent/createRoleComponentInner/permission.js

```diff
@@ -0,0 +1,103 @@
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from "react";
+import { Field, Form } from "react-final-form";
+import { Button, ButtonToolbar, CheckTree } from "rsuite";
+import FormSuite from "rsuite/Form";
+import styles from "../../../../../../../../styles/Settings.module.css";
+import orgRolesData from "../../../data/orgRolesData.json";
+
+/**
+ * 
+ * @param prop - `permissions`, `setPermissions`, `onNext`, `onPrevious`
+ * 
+ * @returns The permission section of create role modal
+ */
+export default function Permission(prop) {
+
+    const { permissions, setPermissions, onNext, onPrevious } = prop;
+
+    const onUpdate = async (values) => {
+
+        onNext();
+        setPermissions(values.permissions);
+    };
+
+    return (
+        <div className={ styles.addUserMainDiv }>
+
+            <div>
+                <Form
+                    onSubmit={ onUpdate }
+                    initialValues={ {
+                        permissions: permissions
+                    } }
+                    render={ ({ handleSubmit, form }) => (
+                        <FormSuite
+                            layout="vertical"
+                            className={ styles.addUserForm }
+                            onSubmit={ event => { handleSubmit(event).then(form.restart); } }
+                            fluid>
+                            <Field
+                                name="permissions"
+                                render={ ({ input }) => (
+                                    <FormSuite.Group controlId="checkbox">
+                                        <FormSuite.Control
+                                            { ...input }
+                                            name="checkbox"
+                                            accepter={ CheckTree }
+                                            data={ orgRolesData }
+                                            defaultExpandItemValues={ [ "/permission" ] }
+                                            cascade
+                                        />
+                                        <FormSuite.HelpText>Assign permission for the role</FormSuite.HelpText>
+                                    </FormSuite.Group>
+                                ) }
+                            />
+
+                            <div className="buttons">
+                                <FormSuite.Group>
+                                    <ButtonToolbar>
+                                        <Button
+                                            className={ styles.addUserButton }
+                                            size="lg"
+                                            appearance="ghost"
+                                            type="submit"
+                                            onClick={ onPrevious }>
```

Comment:
```suggestion
                                            onClick={ onPrevious }
                                        >
```

Check the other places as well.